README.md

```diff
@@ -74,6 +74,18 @@ Metrics Server | Metrics API group/version | Supported Kubernetes version
 
 [Metrics Server releases]: https://github.com/kubernetes-sigs/metrics-server/releases
 
+### High Availability
+
+Latest Metrics Server release can be installed in high availability mode by running:
+
+```shell
+kubectl apply -f https://github.com/kubernetes-sigs/metrics-server/releases/latest/download/high-availability.yaml
+```
+
+Note that this configuration **requires** having a cluster with at least 2 nodes on which Metrics Server can be scheduled.
+
+Also, to maximize the efficiency of this highly available configuration and not waste any resource, it is **recommended** to add the `--enable-aggregator-routing=true` CLI flag to the kube-apiserver so that requests sent to Metrics Server are load balanced between the 2 instances.
```

Comment:
Note, current HA configuration runs two instances with resources enough, basically doubling the resources. setting `--enable-aggregator-routing` might lower the traffic to single instance, but it will not reduce resource overhead.

Still we might recommend it as it should be more reliable in HA mode, but I would like us to conduct some tests before that.

Reply:
Yeah, I wasn't meaning it as it will reduce the resource overhead, but I wanted to emphasize that enabling routing would maximize the efficiency of the configuration and prevent from wasting the resources allocated to the instance that isn't serving metrics.

Should we integrate this configuration into the scale tests or running some basic scale tests locally should be enough?

Reply:
If we are not taking a direct dependency on it (setting resources based on this config), then I think manual testing should be enough.

Reply:
I think we can compare the results of the tests.
so we can verify if running 2 instances can provide benefit of high availability.
- how long apiserver takes to notice failure of replica
- resource consumption

Reply:
@yangjunmyfm192085 Could you do the tests to notice a failure?

For resource consumption I think benefits will be too small to motivate users to change their cluster configuration. On 100 node clusters they will get ~150MB, which is not much for cluster that has 100x8GB. Better way to approach it would be to push this flag as default, so in future releases we can depend on it.

Reply:
# Test enviroment:
- kind deployment kubernetes cluster
- kind version: kind v0.11.0 go1.16.4 linux/amd64
- kubernetes version: v1.21
-  add the `--enable-aggregator-routing=true` CLI flag to the kube-apiserver
- test pods: Use the image and yaml files
 `k8s.gcr.io/metrics-server/metrics-server:v0.5.0`
 `high-availability.yaml`

# Test result:
- 2 replicas of Metrics Server are both serving traffic at time
- The following scenarios, metrics-server service can delete the endpoint of the problem pod in time,so it never has downtime
  - stop kublet
  - kill metrics-server process
  - delete metrics-server pod Manually
- The following scenario,it has downtime occasionally when  metrics-server service can't delete the endpoint of the problem pod in time
  - stop node

Reply:
FWIW, I removed the part mentioning that `it will not waste any resource`.

> The following scenario,it has downtime occasionally when metrics-server service can't delete the endpoint of the problem pod in time

Not sure how the actual routing feature works upstream, but this may be an issue that we could raise and improve?

Reply:
Yeah, this is an issue, we can discuss how to analyse and optimize later.

Reply:
I have done some research about this issue recently, I think the  downtime depend on how fast the controller-manager  discover kubelet does not report the node status.
For this scenario, do we need to note it in the readme file?